flow/hooks/git_workspace_tracking.py

```diff
@@ -0,0 +1,115 @@
+# Copyright (c) 2018 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+from __future__ import print_function
+import os
+import json
+import logging
+from collections import defaultdict
+
+import git
+
+from .git_util import collect_metadata_with_git as collect_metadata
+
+
+logger = logging.getLogger('git_tracking')
+
+GIT_COMMIT_MSG = """{title}
+
+*This commit was auto-generated.*
+"""
+
+
+_WARNINGS = defaultdict(set)
+
+
+def _ignore_hidden_files(root):
+    "Do not commit hidden files to the git repository."
+    with open(os.path.join(root, '.gitignore'), mode='a+') as file:
+        file.seek(0)
+        if '.*\n' not in file.readlines():
+            file.write('.*\n')
```

Comment:
I would add a logger entry or something here (or maybe in the `_get_or_init_git_repo` function`, just update the existing log entry to say that the gitignore file is being initialized to ignore hidden files) to indicate to users that a repo didn't exist, so you're making one and automatically ignoring hidden files.

Reply:
It may also make sense for this to be an option (which is what we discussed the last time around with this PR). However, since it's relatively deep in the call stack and hidden from the user, just notifying the user loudly is probably good enough (and they can change it if they want), rather than trying to provide an API.